src/PrestaShopBundle/Bridge/AddActionInterface.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge;
+
+/**
+ * Define needed constant for addAction methods
+ */
+interface AddActionInterface
+{
+    public const ACTION_TYPE_BULK = 'bulk';
+    public const ACTION_TYPE_ROW = 'row';
+    public const ACTION_TYPE_HEADER_TOOLBAR = 'header_toolbar';
+    public const ACTION_TYPE_LIST_HEADER_TOOLBAR = 'list_header_toolbar';
+
+    public function addAction(string $type, string $action, array $config = []): void;
+}
```

Comment:
This interface looks very "list" specific. What about other page types to migrate ?
Suggestion: define a generic action interface to implement in dedicated classes:
- BulkListAction
- RowListAction
- HeaderToolbarAction
- HeaderToolbarListAction

Much easier to add more actionsin the future. Each implementation might require specific dependencies like ControllerConfiguration, and propose a specific behavior.
```
// In Symfony controller
$this->addAction(
    new BulkListAction(
        $this->controllerConfiguration,
        'actionName',
        [
            // options
        ]
    )
);
```

Reply:
@jolelievre I didn't say all actions were "List" related, only "List" oriented :)
I agree with you, each controller type is responsible of handling its own action types:
- `protected function addBulkListAction(string $name, array $options): void`
- `protected function addRowListAction(string $name, array $options): void`

Although I suggest to use Symfony OptionResolver to handle the options:
```
    protected function addBulkListAction(string $name, array $options): void
    {
        $resolver = new OptionsResolver();
        $resolver->setDefaults([
            // Whatever
        ]);
        $this->controllerConfiguration->bulkActions[$name] = $resolver->resolve($options);
    }
```

Reply:
Thank you for the feedback, I think your proposition is a very good idea!! I will implement this 😄